docs/BuildFromSource.md

```diff
@@ -33,6 +33,7 @@ Building ASP.NET Core on Windows requires:
     ```
 
     However, the build should find any JDK 11 or newer installation on the machine.
+   * Set JAVA_HOME environment variable with the path of java installation directory (Gradle needs this for execution)
```

Comment:
```suggestion
   * Set the JAVA_HOME environment variable with the path of the java installation directory (Gradle needs this for execution)
     * This will be `RepoRoot/.tools/jdk/win-x64/` if you used the `InstallJdk.ps1` script
     * This will be `C:/Program FIles/Java/jdk<version>/` if you installed the JDK globally
```

Reply:
@BrennanConroy note build.ps1 will find Java and set `JAVA_HOME` if you used the InstallJdk.ps1 script.

Reply:
Mostly because this is leading up to https://github.com/dotnet/aspnetcore/blob/master/docs/BuildFromSource.md#user-content-building-on-command-line But, I guess using Visual Studio to run Gradle tests might get dicey.

Reply:
I was trying to build signalR project using build.cmd.
I installed java without InstallJdk.ps1 and it didn't set the JAVA_HOME variable.
Also I don't think even installing JDK with InstallJdk.ps1 will set the JAVA_HOME variable at system level.
Gradle needs either java command or JAVA_HOME env variable.

Reply:
@sandyarmstrong you're correct InstallJdk.ps1 will not set `JAVA_HOME` at the system level. The only thing you're missing is the `build.cmd` scripts all use the root `build.ps1` script and that sets `JAVA_HOME` before starting `msbuild`. Then again, as @BrennanConroy points out, devs may also need `JAVA_HOME` to be set when building in Visual Studio.

Putting his suggestion and mine together:
```suggestion
   * Set the `JAVA_HOME` environment variable with the path of the java installation directory if your installation did not do that automatically. (Gradle needs this for execution.)
      * This will be `RepoRoot/.tools/jdk/win-x64/` if you used the `InstallJdk.ps1` script
      * This will be `C:/Program FIles/Java/jdk<version>/` if you installed the JDK globally
```